File: problem/box/6/main.c
/*
 * main.c: main file for "box" solutions
 *
 * Copyright (C) 2020  Rudy Matela
 * All rights reserved.
 *
 *
 * This file is part of the CSCX Online Judge.
 *
 * This is to be linked to the submitted file.
 * It processes values from standard input then from the "in.txt" file.
 */
#include <stdio.h>
int volume(int, int, int);
int area(int, int, int);
int main()
{
	int w, h, d;
	FILE *f;
	while (scanf(" %d %d %d", &w, &h, &d)==3) {
		printf("The volume of a %d by %d by %d box is %d.\n",w,h,d,volume(w,h,d));
		printf("The surface area of a %d by %d by %d box is %d.\n",w,h,d,area(w,h,d));
		printf("\n");
	}
	f = fopen("in.txt", "r");
	if (!f) {
		fprintf(stderr, "could not open in.txt");
		return 1;
	}
	while (fscanf(f," %d %d %d", &w, &h, &d)==3) {
		printf("The volume of a %d by %d by %d box is %d.\n",w,h,d,volume(w,h,d));
		printf("The surface area of a %d by %d by %d box is %d.\n",w,h,d,area(w,h,d));
		printf("\n");
	}
	fclose(f);
	return 0;
}

File: problem/box/6/main.hs
-- main.hs: main file for "box" solutions
--
-- Copyright (C) 2020  Rudy Matela
-- All rights reserved.
--
--
-- This file is part of the CSCX Online Judge.
--
-- This is to be linked to the submitted file.
-- It processes values from standard input then from the "in.txt" file.
module AltMain (main) where

import Main (volume, area)
import Text.Printf

solve1 :: Int -> Int -> Int -> String
solve1 w h d = printf "The volume of a %d by %d by %d box is %d."       w h d (volume w h d)
    ++ "\n" ++ printf "The surface area of a %d by %d by %d box is %d." w h d (area w h d)
    ++ "\n"

io :: String -> String
io = unlines . map ((\[x,y,z] -> solve1 x y z) . map read . words) . lines

main :: IO ()
main = do
  -- processes standard input
  interact io

  -- processes in.txt
  more <- readFile "in.txt"
  putStr $ io more
